packages/shared/lib/core/utils/crypto/utils/isValidHexAddress.ts

```diff
@@ -0,0 +1,4 @@
+export function isValidHexAddress(text: string): boolean {
+    const isHex = text.startsWith('0x')
+    return isHex && !/^(0x08)?[0-9a-f]{64}?(?:0[1-9]|[1-5][0-9]|6[0-4])?0{8}$/i.test(text)
```

Comment:
Might be good to extract the regexp into a const, i.e.
```typescript
export const HEX_REGEXP = /^(0x08)?[0-9a-f]{64}?(?:0[1-9]|[1-5][0-9]|6[0-4])?0{8}$/i
```

Reply:
We just need to be careful about the `/i` part at the end if we do make it a constant b/c the state is stored in the regexp so as long as we test wrong inputs multiple times back to back, should be alright